packages/orbit-tailwind-preset/src/presets/components/index.ts

```diff
@@ -66,191 +70,194 @@ const getBackgroundColors = () => {
 
 const cfg = (options?: Options): Config => {
   const { disablePreflight = true } = options || {};
+  // make palette colors (foundation colors) defined as css vars and make components tokens inherit it
+  const theme = getTokens(cssVarsFoundation);
+  const componentTokens = getComponentLevelToken(theme);
 
   return {
     content: ["auto"],
-    presets: [orbitFoundationPreset],
+    presets: [orbitFoundationPreset(theme)],
     corePlugins: {
       preflight: disablePreflight ? false : undefined,
     },
     theme: {
+      variables: {
+        // paletteBlueNormal -> --palette-blue-normal
+        DEFAULT: Object.entries(theme).reduce((acc, [key, value]) => {
+          acc[kebabCase(key)] = value;
+          return acc;
+        }, {}),
+      },
       extend: {
         screens: {
-          "sm-mm": { max: px(defaultTokens.breakpointMediumMobile - 1) },
+          "sm-mm": { max: px(theme.breakpointMediumMobile - 1) },
           "mm-lm": {
-            min: px(defaultTokens.breakpointMediumMobile),
-            max: px(defaultTokens.breakpointLargeMobile - 1),
+            min: px(theme.breakpointMediumMobile),
+            max: px(theme.breakpointLargeMobile - 1),
           },
           "lm-tb": {
-            min: px(defaultTokens.breakpointLargeMobile),
-            max: px(defaultTokens.breakpointTablet - 1),
+            min: px(theme.breakpointLargeMobile),
+            max: px(theme.breakpointTablet - 1),
           },
           "tb-de": {
-            min: px(defaultTokens.breakpointTablet),
-            max: px(defaultTokens.breakpointDesktop - 1),
+            min: px(theme.breakpointTablet),
+            max: px(theme.breakpointDesktop - 1),
           },
           "de-ld": {
-            min: px(defaultTokens.breakpointDesktop),
-            max: px(defaultTokens.breakpointLargeDesktop - 1),
+            min: px(theme.breakpointDesktop),
+            max: px(theme.breakpointLargeDesktop - 1),
           },
         },
         fontSize: {
-          "heading-display": defaultTokens.headingDisplayFontSize,
-          "heading-display-subtitle": defaultTokens.headingDisplaySubtitleFontSize,
-          "heading-title1": defaultTokens.headingTitle1FontSize,
-          "heading-title2": defaultTokens.headingTitle2FontSize,
-          "heading-title3": defaultTokens.headingTitle3FontSize,
-          "heading-title4": defaultTokens.headingTitle4FontSize,
-          "heading-title5": defaultTokens.headingTitle5FontSize,
-          "heading-title6": defaultTokens.headingTitle6FontSize,
-          "button-large": defaultTokens.buttonLargeFontSize,
-          "button-normal": defaultTokens.buttonNormalFontSize,
-          "button-small": defaultTokens.buttonSmallFontSize,
-          "form-element-normal": defaultTokens.formElementNormalFontSize,
+          "heading-display": theme.headingDisplayFontSize,
```

Comment:
no. We making our theme preset themable, for now only inside (by calling getTokens), but later it's still has to have an option to do that froom the outside (the same way as with OrbitProvider now)
for Balkan and other projects. 

Plus, when now we have OrbitProvider and css variables changes, it might be needed to have custom foundation with custom typography

TLDR -> in components never should be used defaultTokens, only in foundation preset

Reply:
Ok, I understand the concept of not calling defaultTokens on the components scope, but ultimately that's not necessary right now, or is it? I mean, would it break if we kept calling defaultTokens? Or would it just be wrong on the abstraction of the tokens?

Reply:
I think it will break it with *OrbitProvider*, when they change the theme object. Anyway, I feel like it's the right way and safer to have it like that, plus it's already preparing it for future adjustment 🤔